### DISCUSSION TITLE
How can I solve "No space left on device" error on Milvus server?

### DISCUSSION THREAD

**Discussion by austinjung** [milvus-io/milvus#6490] [Q&A and General discussion]
I got the following error after I ran a stress test of similarity searching.
Could you help me how to config or solve the Milvus server to solve this issue?

Thanks.
`---------------------------------------------------------------------------
BaseException                             Traceback (most recent call last)
<ipython-input-307-c4230e98990b> in <module>
     11         results = t2_collection_in_cpu.search(
     12             embedding['embedding'].tolist(), 'embedding', {}, limit=10, expr=None, partition_name=[partition_name],
---> 13             output_fields=['system_code', 'entity', 'synonym']
     14         )
     15         has_exception = False

/usr/local/lib/python3.6/dist-packages/pymilvus_orm/collection.py in search(self, data, anns_field, param, limit, expr, partition_names, output_fields, timeout, **kwargs)
    625         conn = self._get_connection()
    626         res = conn.search_with_expression(self._name, data, anns_field, param, limit, expr,
--> 627                                           partition_names, output_fields, timeout, **kwargs)
    628         if kwargs.get("_async", False):
    629             return SearchFuture(res)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py in handler(self, *args, **kwargs)
     61                 self._update_connection_pool()
     62             except Exception as e:
---> 63                 raise e
     64             finally:
     65                 counter += 1

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py in handler(self, *args, **kwargs)
     49             counter = 1
     50             try:
---> 51                 return func(self, *args, **kwargs)
     52             except grpc.RpcError as e:
     53                 # DEADLINE_EXCEEDED means that the task wat not completed

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py in inner(self, *args, **kwargs)
     38     @functools.wraps(func)
     39     def inner(self, *args, **kwargs):
---> 40         return func(self, *args, **kwargs)
     41 
     42     return inner

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py in search_with_expression(self, collection_name, data, anns_field, param, limit, expression, partition_names, output_fields, timeout, **kwargs)
   1040         with self._connection() as handler:
   1041             kwargs["_deploy_mode"] = self._deploy_mode
-> 1042             return handler.search_with_expression(collection_name, data, anns_field, param, limit, expression, partition_names, output_fields, timeout, **kwargs)
   1043 
   1044     @retry_on_rpc_failure(retry_times=10, wait=1)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     61                     raise CollectionNotExistException(e.code, e.message)
     62 
---> 63                 raise e
     64             except grpc.FutureTimeoutError as e:
     65                 record_dict["RPC timeout"] = str(datetime.datetime.now())

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     53                     self.ping()
     54                 record_dict["RPC start"] = str(datetime.datetime.now())
---> 55                 return func(self, *args, **kwargs)
     56             except BaseException as e:
     57                 LOGGER.error("Error: {}".format(e))

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     90             raise CollectionNotExistException(ErrorCode.CollectionNotExists,
     91                                               f"collection {collection_name} doesn't exist!")
---> 92         return func(self, *args, **kwargs)
     93 
     94     return handler

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in search_with_expression(self, collection_name, data, anns_field, param, limit, expression, partition_names, output_fields, timeout, **kwargs)
    620         auto_id = collection_schema["auto_id"]
    621         kwargs["auto_id"] = auto_id
--> 622         return self._execute_search_requests(requests, timeout, **kwargs)
    623 
    624     @error_handler(None)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     61                     raise CollectionNotExistException(e.code, e.message)
     62 
---> 63                 raise e
     64             except grpc.FutureTimeoutError as e:
     65                 record_dict["RPC timeout"] = str(datetime.datetime.now())

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in handler(self, *args, **kwargs)
     53                     self.ping()
     54                 record_dict["RPC start"] = str(datetime.datetime.now())
---> 55                 return func(self, *args, **kwargs)
     56             except BaseException as e:
     57                 LOGGER.error("Error: {}".format(e))

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in _execute_search_requests(self, requests, timeout, **kwargs)
    589             if kwargs.get("_async", False):
    590                 return SearchFuture(None, None, True, pre_err)
--> 591             raise pre_err
    592 
    593     def _batch_search(self, collection_name, query_entities, partition_names=None, fields=None, timeout=None, **kwargs):

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py in _execute_search_requests(self, requests, timeout, **kwargs)
    580 
    581                 if response.status.error_code != 0:
--> 582                     raise BaseException(response.status.error_code, response.status.reason)
    583 
    584                 raws.append(response)

BaseException: <BaseException: (code=1, message=IO error: No space left on deviceWhile appending to file: /var/lib/milvus/rdb_data/002773.sst: No space left on device)>
`

**Comment by yhmo** [comment]
The error "IO error: No space left on device" means it is out of storage.
Two possible reason:
1. the hard disk is almost full.
2. the software has a bug to try to write a very huge file.

Firstly please check your local hard disk, how much free space is left.
